Formula/r/recyclarr.rb

```diff
@@ -0,0 +1,33 @@
+class Recyclarr < Formula
+  desc "Automatically sync TRaSH Guides to your Sonarr and Radarr instances"
+  homepage "https://recyclarr.dev"
+  version "7.4.0"
+  url "https://github.com/recyclarr/recyclarr.git", tag: "v#{version}"
+  license "MIT"
+
+  depends_on "dotnet@8" => :build
+  # recyclarr has a dependency on git, which according to brew audit is always available
+
+  def install
+    os = OS.mac? ? "osx" : OS.kernel_name.downcase
+    arch = Hardware::CPU.intel? ? "x64" : Hardware::CPU.arch.to_s
+
+    args = %W[
+      --configuration Release
+      --runtime #{os}-#{arch}
+      --output #{libexec}
+      --self-contained true
```

Comment:
why self-contained?

Reply:
Because I do not want to require dotnet runtime to be installed on systems that install recyclarr. Also it's generally recommended for single-file executable applications to be self contained.

Reply:
I dont think that is acceptable for us.

Reply:
https://docs.brew.sh/Acceptable-Formulae#stuff-that-requires-vendored-versions-of-homebrew-formulae and https://docs.brew.sh/Acceptable-Formulae#shared-vs-static-libraries

Reply:
> [docs.brew.sh/Acceptable-Formulae#stuff-that-requires-vendored-versions-of-homebrew-formulae](https://docs.brew.sh/Acceptable-Formulae#stuff-that-requires-vendored-versions-of-homebrew-formulae) and [docs.brew.sh/Acceptable-Formulae#shared-vs-static-libraries](https://docs.brew.sh/Acceptable-Formulae#shared-vs-static-libraries)

It's not clear to me how the "vendored versions" information directly relates to .NET self contained applications. Could you elaborate?

I'm not sure why Homebrew is being so opinionated about how I build my software. Can you help me understand the objective issue with using a self contained app?

Also from your linked docs it also says:

> Homebrew’s primary mission is to be useful rather than ideologically pure. If we cannot package something without using vendored upstream versions: so be it; better to have packaged software in Homebrew than not.

I'm genuinely interested in working through things but I need to draw a line somewhere. I have yet to see an issue pointed out to me that makes self contained .NET applications not viable.

Reply:
> It's not clear to me how the "vendored versions" information directly relates to .NET self contained applications. Could you elaborate?

By marking the application as self-contained you're vendoring the entire .NET system into the application, which is something that we explicitly disallow.

> I'm not sure why Homebrew is being so opinionated about how I build my software. Can you help me understand the objective issue with using a self contained app?

We're not opinionated about how you build the software, but we do have a strong opinion on how we ship software to users in a way that works for them and for us.

> I'm genuinely interested in working through things but I need to draw a line somewhere. I have yet to see an issue pointed out to me that makes self contained .NET applications not viable.

It does not fix a problem, it makes one worse. You're trying to fix "what if users don't have .NET", and by doing so you're creating "All my apps ship with their own version of .NET, so now I have 6 versions installed". The whole point of a package manager is for it to manage the package you want, and any it depends on.

Reply:
For what it's worth (and I don't suspect it's worth much), but "All my apps ship with their own version of .NET, so now I have 6 versions installed" is a false equivalency; The application isn't installing .NET, so it wouldn't really count as multiple side-by-side installations; other .NET applications would not have access to the dependencies shipped with the application.

I really didn't feel that users needed to pull down a .NET runtime separately just to run a single executable application, but I also don't feel super strongly about it. The most important thing is understanding _why_ the requirement exists. I suppose it's just for the ability to address security vulnerabilities in .NET runtime itself without requiring developers to do the upgrade themselves. The same issue will still exist for package references, which are not part of the .NET runtime, but I suppose some is better than none is the policy you're trying to adopt.

I'm happy to make the change. Thanks for clarifying.

Reply:
Does homebrew support multiple side by side versions of dotnet sdk/runtime? I checked google and found information [here](https://stackoverflow.com/questions/76706036/how-to-install-multiple-dotnet-versions-using-brew) that shows people have to use some unofficial workarounds to get this.

I just want to make sure that if I list a dependency on `dotnet@8` that this won't conflict with other dotnet versions (e.g. version 9) that users already have installed.

Thank you.

Reply:
Whatever dependencies your tool has will be managed by brew, which can install multiple versions of dotnet but as a policy only installs supported/major versions